### PR TITLE
Event Filter

### PR DESCRIPTION
**Pull request by felixrindt**
* [x] rebuild calendar view
* [x] fix TODOs
* [x] check API compatibility
  * [x] while at it, maybe add a limit to the events API endpoint.
* [x] Pagination
* [x] remove past event view
* [x] calendar view must be scrolled on mobile if filter is always visible... what to do?
* [x] update and add tests

move to different issue/PR

* match this filters design with the modellogging filter -> seperate issue #860
  * specifically: the select2 inputs don't look like native bootstrap5 selects, so theres some theming broken
* remove past event permission -> seperate make-migrations PR
* #472 
* why can't I change the event type of an event later on? #577

### CODE REVIEW

**Comment by jeriox** [comment]
regarding the modellogging filter you could also think about https://github.com/ephios-dev/ephios/issues/418

File: .env.example [hunk 1]
```diff
@@ -11,3 +11,4 @@ EMAIL_URL=dummymail://
 DEFAULT_FROM_EMAIL=webmaster@localhost
 SERVER_EMAIL=root@localhost
 ADMINS=Root User <root@localhost.dev>
+# VAPID_PRIVATE_KEY_PATH=data/private_key.pem # push notifications, create in `data/` with `vapid --gen`
```
Comment (felixrindt) [inline .env.example:14]: placeholder comment: remember to add the issues from the PR description

Comment (jeriox) [reply .env.example:14]: the last one already exists (#577)

**Comment by jeriox** [comment]
grafik
when first opening the calender view, I find it quite unintuitive where to look for the navigation. I expected them to be visible somewhere and needed to remind myself that they are inside of the filter. I think users will struggle with that, especially since they have been existing before

**Review by jeriox — APPROVED** [review]

**Pull request merged**